File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/smart_pointer_member_runme.m
smart_pointer_member

f = Foo();
f.y = 1;

if (f.y != 1)
  error
endif

b = Bar(f);
b.y = 2;

if (f.y != 2)
  error("f.y = %i, b.y = %i",f.y,b.y)
endif

if (swig_this(b.x) != swig_this(f.x))
  error
endif

if (b.z != f.z)
  error
endif

try
  if (Foo.z == Bar.z)
    error
  endif
    error
catch
end_try_catch

  

File: support_tools/swigwin-3.0.12/Examples/test-suite/python/smart_pointer_member_runme.py
from smart_pointer_member import *


def is_new_style_class(cls):
    return hasattr(cls, "__class__")

f = Foo()
f.y = 1

if f.y != 1:
    raise RuntimeError

b = Bar(f)
b.y = 2

if f.y != 2:
    print f.y
    print b.y
    raise RuntimeError

if b.x != f.x:
    raise RuntimeError

if b.z != f.z:
    raise RuntimeError

if is_new_style_class(Bar):  # feature not supported in old style classes
    if Foo.z == Bar.z:
        raise RuntimeError
